File: frontend/src/app/models/tournament/tournament-item.response.ts
import { UserRef } from "../common/user-ref.model";

export interface TournamentItemResponse {
    id: number;
    name: string;
    startDate: string;
    endDate: string;
    location? : string | null;
    status: 'UPCOMING' | 'ONGOING' | 'FINISHED';
    system?: 'GROUP' | 'BRACKET' | 'MIXED';
    organizer: UserRef;
    publicVisible: boolean;
    followed: boolean;
}

File: src/main/java/com/tournament/app/footycup/backend/dto/tournament/TournamentItemResponse.java
package com.tournament.app.footycup.backend.dto.tournament;

import com.tournament.app.footycup.backend.dto.common.UserRef;

import java.time.LocalDate;

public record TournamentItemResponse(
        Long id,
        String code,
        String name,
        LocalDate startDate,
        LocalDate endDate,
        String location,
        String status,
        UserRef organizer,
        boolean publicVisible,
        boolean followed
) {
}
